packages/libs/client-utils/src/marmalade/create-token-id.ts

```diff
@@ -1,50 +1,68 @@
-import type { ChainId } from '@kadena/client';
-import { Pact } from '@kadena/client';
-import { dirtyReadClient } from '../core';
+import type {
+  BuiltInPredicate,
+  ChainId,
+  IPactModules,
+  PactReference,
+  PactReturnType,
+} from '@kadena/client';
+import { Pact, readKeyset } from '@kadena/client';
+import { addKeyset, execution } from '@kadena/client/fp';
+import type { IPactInt, NetworkId } from '@kadena/types';
+import { pipe } from 'ramda';
+import { dirtyReadClient } from '../core/client-helpers';
 import type { IClientConfig } from '../core/utils/helpers';
+import type { ICreateTokenPolicyConfig } from './config';
+import { validatePolicies } from './helpers';
 
 interface ICreateTokenIdInput {
+  policyConfig?: ICreateTokenPolicyConfig;
+  policies?: string[];
   uri: string;
-  policies: string[];
-  account: { account: string; publicKeys: string[] };
-  creationGuard: { keys: string[]; pred: 'keys-all' | 'keys-2' | 'keys-any' };
-  gasPayer: { account: string; publicKeys: string[] };
+  precision: IPactInt | PactReference;
+  creator: {
+    keyset: {
+      keys: string[];
+      pred: BuiltInPredicate;
+    };
+  };
   chainId: ChainId;
+  networkId: NetworkId;
+  host?: IClientConfig['host'];
 }
 
-/**
- * @alpha
- */
-function createTokenIdCommand({
-  account,
-  creationGuard,
-  gasPayer = account,
-  chainId,
+export const createTokenId = ({
+  policies = [],
   uri,
-  policies = [
-    'marmalade-v2.non-fungible-policy-v1',
-    'marmalade-v2.guard-policy-v1',
-  ],
-}: ICreateTokenIdInput) {
-  const transaction = Pact.builder
-    .execution(
-      `(marmalade-v2.ledger.create-token-id {"precision": 0, "policies": (read-msg "policies"), "uri": (read-string "uri")} (read-keyset "creation_guard"))`,
-    )
-    .addData('uri', uri)
-    .addData('policies', policies)
-    .addData('creation_guard', creationGuard)
-    .addSigner(gasPayer.publicKeys, (signFor) => [signFor('coin.GAS')])
-    .setMeta({ chainId, senderAccount: gasPayer.account })
-    .getCommand();
-  return transaction;
-}
-
-// turbo run build --filter @kadena/kadena-cli^...
+  precision,
+  creator,
+  policyConfig,
+  chainId,
+  networkId,
+  host,
+}: ICreateTokenIdInput) => {
+  validatePolicies(policyConfig, policies);
 
-/**
- * @alpha
- */
-export const createTokenId = (
-  inputs: ICreateTokenIdInput,
-  config: Omit<IClientConfig, 'sign'>,
-) => dirtyReadClient(config)(createTokenIdCommand(inputs));
+  return pipe(
+    () =>
+      Pact.modules['marmalade-v2.ledger']['create-token-id'](
+        {
+          precision,
+          uri,
+          policies: () =>
+            policies.length > 0 ? `[${policies.join(' ')}]` : '[]',
+        },
```

Comment:
Does this work when tested with policies?

Reply:
Yeah, I have it also covered in a test here https://github.com/kadena-community/kadena.js/pull/1484/files#diff-c7b784837590ebd192b560bcca5c3bcd40cf90e640d56e4a6c166bf5fec1ae38R109